src/main/java/org/glassfish/build/ExecMojo.java

```diff
@@ -102,10 +103,22 @@ && new File(executable + ".bat").exists()) {
         getLog().info("executable: " + executable);
         exec.createArg().setLine(commandlineArgs);
         exec.setFailonerror(failOnError);
+        String javaHome = System.getenv("JAVA_HOME");
+        if (javaHome != null) {
+            exec.addEnv(getVariable("AS_JAVA", javaHome));
+            exec.addEnv(getVariable("JAVA_HOME", javaHome));
+        }
         getLog().info("commandLineArgs: " + commandlineArgs);
         exec.execute();
     }
 
+    private Variable getVariable(final String key, final String value) {
```

Comment:
I'd rather see [`static`] `variable(...` or `createVariable(...` as now the call `getVariable("N", "V")` sounds for me as `getProperty` with `"V"` being default.
But it's detail.

Reply:
Eh, I did not push it, I had the same idea, thanks! :D